op-supervisor/supervisor/backend/cross/hazard_set.go

```diff
@@ -0,0 +1,188 @@
+package cross
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/depset"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// MaxHazardBlockChecks is the maximum number of blocks that can be processed when building a hazard set.
+// It is a safety limit to prevent potential infinite loops or excessive resource consumption.
+const MaxHazardBlockChecks = 10000
```

Comment:
Is this a reasonable value for a stop-limit to prevent run-away looping? We shouldn't expect to exhaust the limit in normal operation.

Reply:
> if there's any possibility that it can be violated

If this limit is never going to be hit, then we should remove it. And if the limit *is* going to be hit, we should remove it (so the op-program is correct).

A supervisor which has stalled because it is somehow in an infinite hazard-checking loop is equally as healthy as a node that is halted because it hit the `MaxHazardBlockChecks`.